src/Vote.Monitor.Api.Feature.Observer/Import/Endpoint.cs

```diff
@@ -1,23 +1,84 @@
+    private readonly IRepository<ImportValidationErrors> _errorRepo;
+    private readonly ICsvParser<ObserverImportModel> _parser;
+    private readonly ILogger<Endpoint> _logger;
 
-    public Endpoint(IReadRepository<ObserverAggregate> repository)
+    public Endpoint(IRepository<ObserverAggregate> repository,
+        IRepository<ImportValidationErrors> errorRepo,
+        ICsvParser<ObserverImportModel> parser,
+        ILogger<Endpoint> logger)
     {
         _repository = repository;
+        _errorRepo = errorRepo;
+        _parser = parser;
+        _logger = logger;
     }
 
     public override void Configure()
     {
         Post("/api/observers:import");
         DontAutoTag();
         Options(x => x.WithTags("observers"));
+        AllowFileUploads();
     }
 
-    public override async Task<Results<Ok<ObserverModel>, NotFound>> ExecuteAsync(Request req, CancellationToken ct)
+    public override async Task<Results<NoContent, BadRequest<ImportValidationErrorModel>>> ExecuteAsync(Request req, CancellationToken ct)
     {
-        throw new NotImplementedException();
+        var parsingResult = _parser.Parse(req.File.OpenReadStream());
+        if (parsingResult is ParsingResult2<ObserverImportModel>.Fail failedResult)
+        {
+            StringBuilder errors = new StringBuilder("");
+            foreach (var validationFailure in failedResult.ValidationErrors.SelectMany(x => x.Errors))
+            {
+                errors.AppendLine(validationFailure.ErrorMessage);
+            }
+            string csv = CsvRowParsedHelpers<ObserverImportModel>.ConstructErrorFileContent(failedResult.Items);
+            var errorSaved = await _errorRepo.AddAsync(new(ImportType.Observer, req.File.Name, csv, DateTime.Now), ct);
+            return TypedResults.BadRequest(new ImportValidationErrorModel { Id = errorSaved.Id, Message = errors.ToString() });
+
+        }
+
+        var importedRows = parsingResult as ParsingResult2<ObserverImportModel>.Success;
+        //if (importedRows == null) 
+        //{
+        //    ThrowError("No rows imported");
+        //}
+        List<ObserverAggregate> observers = importedRows!
+            .Items
+            .Select(x => new ObserverAggregate(x.Name, x.Email, x.Password, x.PhoneNumber))
+            .ToList();
+        List<ObserverAggregate> observersToAdd = new();
+        foreach (var obs in observers)
+        {
+            var specification = new GetObserverByLoginSpecification(obs.Login);
+            var hasObserverWithSameLogin = await _repository.AnyAsync(specification, ct);
+
+            if (hasObserverWithSameLogin)
+            {
+                _logger.LogWarning("An Observer with email {obs.Login} already exists!", obs.Login);
+            }
+            else
+            {
+                observersToAdd.Add(obs);
+            }
+
+        }
+
+        if (observersToAdd.Count > 0) await _repository.AddRangeAsync(observersToAdd, ct);
```

Comment:
Short: Your request is not valid because the data model implemented does not support BulkInsert. 

Long: Adding the code I got the following exception: 
        System.InvalidOperationException: An exception was thrown while attempting to evaluate a LINQ query parameter expression. See the inner exception for more information. To show additional information call 'DbContextOptionsBuilder.EnableSensitiveDataLogging'.
--->System.InvalidOperationException: Using 'FromSqlRaw' on DbSet of 'Observer' is not supported since 'Observer' is part of hierarchy and does not contain a discriminator property.
at Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.GenerateFromSqlQueryRoot(IQueryable source, String sql, Object[] arguments, String memberName)
at Microsoft.EntityFrameworkCore.RelationalQueryableExtensions.FromSqlRaw[TEntity](DbSet`1 source, String sql, Object[] parameters)

  Adding a discriminator to the model will lead to another exception: 

'The mapping strategy 'TPT' specified on 'ApplicationUser' is not supported for entity types with a discriminator.'